dandiapi/api/models/version.py

```diff
@@ -203,7 +207,7 @@ def strip_metadata(cls, metadata):
         ]
         return {key: metadata[key] for key in metadata if key not in computed_fields}
 
-    def _populate_metadata(self, version_with_assets: Version = None):
+    def _populate_metadata(self, version_with_assets: Version | None = None):
```

Comment:
Note: ha, I was expecting `Optional[Version]` but apparently it isn't the convention in dandi-archive:

```shell
(git)lena:~/proj/dandi/dandi-archive[master]git
$> git grep 'Optional\['
(dev3) 1 34993 ->1.....................................:Tue 05 Jul 2022 03:51:59 PM EDT:.
(git)lena:~/proj/dandi/dandi-archive[master]git
$> git grep '| None'    
dandiapi/api/copy.py:    def copy_range(self) -> str | None:
dandiapi/api/mail.py:def build_message(subject: str, message: str, to: list[str], html_message: str | None = None):
dandiapi/api/models/dandiset.py:    def identifier(self) -> str | None:
dandiapi/api/models/upload.py:    def object_key(upload_id, dandiset: Dandiset | None = None):  # noqa: N805
dandiapi/api/models/upload.py:    def object_key(upload_id, dandiset: Dandiset | None = None):
dandiapi/api/models/zarr.py:    def checksum(self) -> str | None:
dandiapi/api/tasks/zarr.py:def yield_files(bucket: str, prefix: str | None = None):
dandiapi/api/tasks/zarr.py:    def read_checksum_file(self) -> ZarrChecksumListing | None:
dandiapi/api/views/asset.py:        glob_pattern: str | None = serializer.validated_data.get('glob')
dandiapi/api/zarr_checksums.py:    def read_checksum_file(self) -> ZarrChecksumListing | None:
```
although according to  https://docs.python.org/3/library/typing.html#typing.Optional  it is only a recent "addition"

```
Changed in version 3.10: Optional can now be written as X | None. 
```

Reply:
You're right, since we only run 3.9, this wouldn't usually work, but we have the following line at the top of the file

```
from __future__ import annotations
```

which allows that syntax.